tests/unit/internal-session-test.js

```diff
@@ -776,6 +776,17 @@ describe('InternalSession', () => {
               done();
             });
           });
+
+          it('it does neither raise an assertion nor an error', function() {
+            let triggered = false;
+            sinon.stub(session, '_lookupAuthenticator').returns(authenticator);
+            session.one('sessionInvalidationFailed', () => (triggered = true));
+
+            session.invalidate().then((val) => {
+              expect(triggered).to.be.false;
```

Comment:
I don't think this assertion is needed actually - only asserting on the fact that a resolved promise is returned should be sufficient. That test would already fail without the above changes as the assertion would be triggered and raise an error which fails the test.

Reply:
As we didn't have a test case beforehand that ensures that no error is thrown when the `invalidate` method is called on an unauthenticated session, I think this is still useful and I can also add an assertion for the return value of the method as you suggested. Do you think this is good or did I miss something?

Reply:
Hm, yeah - I guess what I said above doesn't really make sense

> That test would already fail without the above changes as the assertion would be triggered and raise an error which fails the test.

I'd split this into 2 test cases though - `test('it returns a resolved promise')`, `test('it does not trigger "sessionInvalidationFailed"')` as those are 2 different aspects that don't necessarily need to be tested together in one test case.

Reply:
Yes, sounds good 👍